### DISCUSSION TITLE
Challenging of after subsetting unrelated biological replicates.

### DISCUSSION THREAD

**Discussion by jenny8030** [satijalab/seurat#8308] [General]
Hi all,
I wanted to express my sincere gratitude for the exceptional tool, Seurat, and the ongoing efforts in maintaining it. 
I have encountered some issues while analyzing my data and would like to discuss them with all Seurat users.

**- Information about my data** 
 I have triplicates for each condition: Normal (N-rep1, N-rep2, N-rep3), Disease (D-rep1, D-rep2, D-rep3), and Treatment (T-rep1, T-rep2, T-rep3), totaling 9 samples. Using all of these 9 data,  pre-processing and finding clusters were done.

**- Issue**
 However, I observed that the histology of T-rep1 is similar to the Disease model. Consequently, I decided to exclude the T-rep1 sample from further analysis, such as finding DEG between the Disease and Treatment. I accomplished this by subsetting the T-rep1 sample in my Seurat object using the provided code.
```
subset_seurat_obj <- subset(seurat_obj, subset = Sample != "T-rep1") 
```

**- Question**
 In the process of subsetting:

1. I believe it's unnecessary to re-preprocess this subsetted object(_subset_seurat_obj_), including steps like normalization, finding variable features, sctransform, harmony, running UMAP, finding neighbors, and finding clusters. Am I correct?

2. However, for subsequent analysis, such as visualizing DEG through a heatmap, it's essential to run ScaleData on the subsetted object. What are your thoughts on this approach?

I would greatly appreciate it if the Seurat team could share their insights, and if Seurat users could also contribute by sharing their experiences.

Thank you for your time and support.

Best regards,

------------------------------------------------------------------------------------------------------------------------------

**cf) - My worklow is as follows:**

1. Create a Seurat object for each sample, resulting in 9 objects.
2. Merge the 9 objects.
```
unique(merged_seurat$Condition)    # Normal Disease Treatment
unique(merged_seurat$Sample)        # N-rep1 N-rep2 N-rep3 D-rep1 D-rep2 D-rep3 T-rep1 T-rep2 T-rep3 
```
4. Filter out low-quality cells.
5. Apply normalization, FindVariableFeatures, ScaleData to the merged object (generating the "RNA" assay), and additionally, apply SCTransform (generating the "SCT" assay). Run PCA on the SCT assay.
6. Perform Harmony set grouping variable as Sample
```
harmonized_seurat <- RunHarmony(filtered_seurat, 
                              group.by.vars = c("Sample"), 
                              reduction = "pca", assay.use = "SCT", reduction.save = "harmony", project.dim = F)
```
7. Run UMAP, FindNeighbors, and FindClusters.
```
harmonized_seurat <- RunUMAP(harmonized_seurat, reduction = "harmony", assay = "SCT", dims = 1:set_Harmony) %>% 
  FindNeighbors(reduction = "harmony", dims = 1:set_Harmony) %>% 
  FindClusters(resolution = res)
```

**Comment by samuel-marsh** [comment]
Hi,

Not member of dev team but hopefully can be helpful.  So to answer your question yes you should re-run pipeline after removing object.  The primary reason has to do with the selection of variable features.  Once you remove cells it may change the selection of variable features which were being driven by those cells previously.  This will in turn change all of the downstream aspects as well.  Therefore, you want to re-run so that no part of your analysis reflects effects of those removed cells.

For your second question yes you need to run `ScaleData` for plotting on heatmap and this also needs to be re-run after removing cells because it will change the values.

As this is not a bug I will also move this issue to the discussions section in case other users would like to chime in.

Best,
Sam
